test/trains.js

```diff
@@ -0,0 +1,64 @@
+
+const chai = require( 'chai' )
+const chaiAsPromised = require( 'chai-as-promised' )
+const expect = chai.expect
+const assert = chai.assert
+const db = require( '../db/config' ).db
+const Train = require( '../db/commands/train' )
+
+chai.use( chaiAsPromised )
+
+describe('Train', function() {
+  let firstTrain, secondTrain
+
+  beforeEach( function() {
+    return Promise.all([
+      db.query( 'TRUNCATE trains' ),
+      firstTrain = new Train( { trainNumber: 1, currentStation: 'Downtown', nextStation: 'Elm Street' } ),
```

Comment:
Are `firstTrain` and `secondTrain` a promise? If not they dont belong in a Promise.all

I'd do it like this:

```js
beforeEach(function() {
  return db.query('TRUNCATE trains')
    .then(_ => {
      firstTrain = new Train({
        trainNumber: 1, 
        currentStation: 'Downtown', 
        nextStation: 'Elm Street',
      }),
      secondTrain = new Train({
        trainNumber: 2, 
        currentStation: 'Annex', 
        nextStation: '10th Ave',
      })
      return Promise.all([
        firstTrain.save(),
        secondTrain.save(),
      ])
    })
  ])
})
```